R/namespace_linter.R

```diff
@@ -51,27 +51,10 @@ namespace_linter <- function(check_exports = TRUE, check_nonexports = TRUE) {
       return(lints)
     }
 
-    ## Case 2 (rare?): pkg namespace is broken in pkg::foo
+    ## Case 2/3/4: problems with foo in pkg::foo / pkg:::foo
 
     # run here, not in the factory, to allow for run- vs. "compile"-time differences in package structure
     namespaces <- lapply(packages, function(package) tryCatch(getNamespace(package), error = identity))
-    failed_namespace <- vapply(namespaces, inherits, "condition", FUN.VALUE = logical(1L))
```

Comment:
We should either

1 . also remove `tryCatch()` and just do `lapply(packages, getNamespace)`
2. keep the `failed_namespace` check and throw an error `if (any(failed_namespace))`

Reply:
Good idea. Done. See if the error message is okay or needs to be modified.

Reply:
sorry that was a 1 or 2. how can namespaces inherit from error if we remove tryCatch?

Reply:
🤦 

Sorry for the autopilot mode. 

I think option 2 is better, so going with that one.